### ISSUE TITLE
`helpers.getAuthToken` hangs scripts indefinitely on failed authentication

### ISSUE THREAD

**Issue by johnboxall** [SalesforceCommerceCloud/commerce-sdk#108]
While testing out the SDK [before I had my shortCode](https://github.com/SalesforceCommerceCloud/commerce-sdk/issues/105
), I observed failed authentication would "hang" scripts, leaving Node running indefinitely rather than terminating as expected:

```javascript
// Running this script will result in a HTTP 403 / Forbidden
const { helpers } = require("commerce-sdk");

const parameters = {
  clientId: "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
  organizationId: `f_ecom_aaaa_001`,
  siteId: "RefArch",
  shortCode: "staging-001"
};

helpers
  .getAuthToken({
    parameters,
    body: {
      type: "guest"
    }
  })
  .catch(error => {
    console.error(error);
  });
```

The error prints, but the scripts continues to run until you force kill it. 

I expect it to exit as all promises should be rejected/resolved and all callbacks fired.

**Environment**

* Node v12.16.1
* commerce-sdk ^1.3.0-alpha.6

**Comment by jbegleiter-sf** [comment]
Hey @johnboxall , I've seen this behavior before in node apps when executed from the command line. The event loop simply stays open unless otherwise told to exit. Try dropping in a finally block that exits to this code, e.g.:

```helpers
  .getAuthToken({
    parameters,
    body: {
      type: "guest"
    }
  })
  .catch(error => {
    console.error(error);
  })
  .finally(() => {
    process.exit(0);
  });```

We'll discuss internally how we want to deal with this.

**Comment by johnboxall** [comment]
Thanks!

I think we're on the same page – to be clear, my expectation (as set by experiences with other libraries) would be that the event loop would be clear and execution would automatically end w/o additional handling.

For example, using `node-fetch`:

```bash
// Run using node, this exits cleanly, as expected
const fetch = require('node-fetch')

fetch('https://httpbin.org/status/403')
    .then(res => {
        if (!res.ok) {
            const error = new Error(res.status)
            error.response = res
            throw error
        }
    })
    .catch(error => {
        console.error(error)
    })
```

**Comment by jbegleiter-sf** [comment]
Hey John,

Totally agree, leaving this open as a bug and will discuss with the team.

Cheers,
Josh

**Comment by jbegleiter-sf** [comment]
Hey John,

We are now actively working on a resolution for this bug. I will update this thread when a fix becomes available.

Cheers,
Josh

**Comment by gkhunger-sf** [comment]
Hi John,

The PR to fix this issue got merged yesterday. The fix will be a part of the upcoming release. Here's the PR:

https://github.com/SalesforceCommerceCloud/commerce-sdk/pull/122

Cheers,
Gaurav

**Comment by bredmond-sf** [comment]
Hi @johnboxall, this change has been published as part of 1.3.0-alpha.9 and is now available
https://github.com/SalesforceCommerceCloud/commerce-sdk/releases/tag/v1.3.0-alpha.9
The change disables keep-alive by default (node was waiting for the connection to timeout and that was the reason it wasn't exiting). You weren't seeing similar behavior in your test with node-fetch because node-fetch doesn't include any keep-alive functionality.
